odxtools/odxlink.py

```diff
@@ -26,7 +26,11 @@ def __eq__(self, other: Any) -> bool:
         return self.doc_name == other.doc_name
 
     def __hash__(self) -> int:
-        return hash(self.doc_name) + hash(self.doc_type)
+        return hash(self.doc_name)
+
+    def __equals__(self, other: "OdxDocFragment") -> bool:
+        return self.doc_name == other.doc_name and (self.doc_type is None and other.doc_type is None
+                                                    or self.doc_type == other.doc_type)
```

Comment:
This can cause issues, since you can have A == B, and B == C, but A != C

Reply:
I know; this is why I described this as a hack in the PR description. The intention is to be able to is `OdxDocFragment` objects as keys in dictionaries and getting a result if the document type is unspecified for the object used as index (a la `self._db[doc_frag]` in `OdxLinkDatabase`. if multiple document fragments using the same name but different document types exist, I'm willing to let the result to be undefined). The "clean" solution would be to to use a custom comparison function for these dictionaries, but I have not found a way to do this in a satisfactory manner...

Reply:
okay; it turns out that a custom comparison function is not necessary here: the spec mandates that the DOCREF and the DOCTYPE attributes are both either present or not present in references and if they are not present, we have to use the "local" document fragments where the reference is located. In conjunction this means that we get away with requiring that the `doc_type` attribute is always present and always including it in comparisons.

(also, overloading comparisons works by redefining `__eq__()`, not `__equals__()`. I have no idea what hit me when I did this.)

anyway, [fixed](https://github.com/mercedes-benz/odxtools/pull/322/commits/5e1079731e365a752aa07a165e13787b1924696b). do you see any further issues with this PR @kayoub5?